Build Failed on MAC

Build failed on Mac, and the errors are as follows:

```
Scanning dependencies of target extern_grpc
[  2%] No configure step for 'extern_grpc'
[  2%] Performing build step for 'extern_grpc'
sh: HAS_SYSTEM_PROTOBUF=false: No such file or directory
make[2]: *** [/Users/baidu/.cache/third_party/grpc/src/extern_grpc-stamp/extern_grpc-build] Error 127
make[1]: *** [CMakeFiles/extern_grpc.dir/all] Error 2
make: *** [all] Error 2
```

It seems that the settings here https://github.com/PaddlePaddle/Paddle/pull/5982/files#diff-2a53cba3b34e6f12d648b60ac0b5cc8cR45 leave the MAC BUILD_CMD out since the following BUILD_CMD for MAC and others:

```
IF(APPLE)
  SET(BUILD_CMD make -n | sed "s/-Werror//g" | sh)
ELSE()
  SET(BUILD_CMD make)
ENDIF()
```

--- a/cmake/external/grpc.cmake
+++ b/cmake/external/grpc.cmake
@@ -24,9 +24,9 @@ SET(GRPC_INSTALL_DIR ${THIRD_PARTY_PATH}/install/grpc)
 SET(GRPC_INCLUDE_DIR "${GRPC_INSTALL_DIR}/include/" CACHE PATH "grpc include directory." FORCE)
 SET(GRPC_CPP_PLUGIN "${GRPC_INSTALL_DIR}/bin/grpc_cpp_plugin" CACHE FILEPATH "GRPC_CPP_PLUGIN" FORCE)
 IF(APPLE)
-  SET(BUILD_CMD make -n | sed "s/-Werror//g" | sh)
+  SET(BUILD_CMD make -n HAS_SYSTEM_PROTOBUF=false -s -j8 static grpc_cpp_plugin | sed "s/-Werror//g" | sh)
 ELSE()
-  SET(BUILD_CMD make)
+  SET(BUILD_CMD make HAS_SYSTEM_PROTOBUF=false -s -j8 static grpc_cpp_plugin)
 ENDIF()
 
 ExternalProject_Add(
@@ -42,7 +42,7 @@ ExternalProject_Add(
     # Disable -Werror, otherwise the compile will fail in MacOS.
     # It seems that we cannot configure that by make command.
     # Just dry run make command and remove `-Werror`, then use a shell to run make commands
-    BUILD_COMMAND  ${BUILD_CMD} HAS_SYSTEM_PROTOBUF=false -s -j8 static grpc_cpp_plugin
+    BUILD_COMMAND  ${BUILD_CMD}
     INSTALL_COMMAND make prefix=${GRPC_INSTALL_DIR} install
 )
 

